src/Exercism.TestRunner.CSharp/TestResult.cs

```diff
@@ -1,21 +1,125 @@
+        public string Test { get; }
+        public string Expected { get; }
+        public string Command { get; }
         public string Message { get; }
         public string Output { get; }
         public TestStatus Status { get; }
 
-        private TestResult(string name, TestStatus status, string message, string output) =>
-            (Name, Message, Status, Output) = (name, message, status, output);
+        private TestResult(string name, string test, string expected, string command, TestStatus status, string message, string output) =>
+            (Name, Test, Expected, Command, Message, Status, Output) = (name, test, expected, command, message, status, output);
 
-        public static TestResult FromPassed(ITestPassed test) =>
-            new TestResult(test.TestCase.DisplayName, TestStatus.Pass, null, test.Output);
+        public static TestResult FromPassed(ITestPassed test, SyntaxNode syntaxNode) =>
+            new TestResult(ToName(test.TestCase), ToTest(test.TestCase), ToExpected(syntaxNode), ToCommand(syntaxNode), TestStatus.Pass, null, test.Output);
 
-        public static TestResult FromFailed(ITestFailed test) =>
-            new TestResult(test.TestCase.DisplayName, TestStatus.Fail, TestRunMessage.FromMessages(test.Messages), test.Output);
+        public static TestResult FromFailed(ITestFailed test, SyntaxNode syntaxNode) =>
+            new TestResult(ToName(test.TestCase), ToTest(test.TestCase), ToExpected(syntaxNode), ToCommand(syntaxNode), TestStatus.Fail, TestRunMessage.FromMessages(test.Messages), test.Output);
+        
+        private static string ToName(ITestCase testCase) => testCase.TestMethod.Method.Name.Humanize();
+
+        private static string ToTest(ITestCase testCase) => testCase.DisplayName;
+
+        private static string ToExpected(SyntaxNode syntaxNode)
+        {
+            var assertExpression = syntaxNode.DescendantNodes()
+                .OfType<InvocationExpressionSyntax>()
+                .FirstOrDefault(invocationExpression => 
+                    invocationExpression.Expression is MemberAccessExpressionSyntax memberAccessExpression &&
+                    memberAccessExpression.Expression is IdentifierNameSyntax identifierName &&
+                    identifierName.Identifier.Text == "Assert");
+
+            if (assertExpression == null)
+                return null;
+
+            var memberAccessExpression = (MemberAccessExpressionSyntax)assertExpression.Expression;
+            var assertionIdentifierName = (IdentifierNameSyntax)memberAccessExpression.Name;
+            
+            if (assertionIdentifierName.Identifier.Text == "True")
+                return "true";
+            
+            if (assertionIdentifierName.Identifier.Text == "False")
+                return "false";
+            
+            if (assertionIdentifierName.Identifier.Text == "Equal")
+                return assertExpression.ArgumentList.Arguments[0].ToString();
```

Comment:
It would be worth running through all the existing exercises' tests to make sure people have got the left/right the correct way round when asserting equality.

Reply:
We have done that consistently, so that's nice!